lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,54 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::core::ILogger &log, io::I2c &i2c, const std::uint8_t channel)
+    : log_(log),
+      i2c_(i2c),
+      channel_(channel)
+{
+}
+
+std::optional<std::int16_t> Temperature::read()
+{
+  const auto status_check_result = i2c_.readByte(kTemperature, kStatus);
+  if (!status_check_result) {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature could not read status");
```

Comment:
`logger_log.(...., "Failed to read temerature sensor status at channel %d", channel_); ` -> and similarly for the rest of the logs. Basically just want to standardize logging, so if it's something that went wrong, just start with "Failed to ..." and if something went right start with "Successfully ..." (so you have to think less about the message itself). The `channel_` id in the log just helps identify which exact sensor it was that failed.